tests/drivers/flash_simulator/src/main.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <flash.h>
+#include <device.h>
+
+/* configuration derived from DT */
+#define CONFIG_FLASH_SIMULATOR_BASE_OFFSET DT_FLASH_BASE_ADDRESS
```

Comment:
These should not be `CONFIG_` prefix since these aren't related to any Kconfig options.